File: Java/Basics/ClassTypeTest/Investment.java
class Investment {

	//instance fields
	private double sum;
	private int years;
	private boolean risk;

	//parameterized constructor
	public Investment(double amount, int duration) {
		sum = amount;
		years = duration;
		risk = false;
	}

	//instance methods

	public double getSum() {
		return sum;
	}

	public int getYears() {
		return years;
	}	

	public void allowRisk(boolean yes) {
		risk = yes;
	}

	public double income() {
		float rate = risk ? 8 : 5;
		double amount = sum * Math.pow(1 + rate / 100, years);
		return amount - sum;
	}
}

File: DotNet/Basics/StructTypeTest/DemoApp/Investment.cs
using System;

//user-defined value type
struct Investment
{
    //instance fields
    //readonly field can only be assigned in the constructor
    public readonly double Sum;
    public readonly double Years;
    private bool risk;

    //struct construct must be defined with at least one non-optional parameter
    public Investment(double amount, int period)
    {
        Sum = amount;
        Years = period;
        risk = false;
    }

    public void AllowRisk(bool yes)
    {
        risk = yes;
    }

    public double Income()
    {
        float rate = risk ? 7 : 5;
        double amount = Sum * Math.Pow(1 + rate / 100, Years);
        return amount - Sum;
    }
}
